File: articles/decision-quality-and-strategic-alignment/python/run_all_decision_quality_alignment_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Decision Quality and Strategic Alignment."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "decision_quality_score.py",
    "strategic_alignment_score.py",
    "implementation_readiness.py",
    "alignment_drift_analysis.py",
    "adaptive_performance_simulation.py",
    "decision_review_queue.py",
    "decision_record_exporter.py",
    "decision_quality_strategic_alignment_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python decision quality and alignment workflows completed.")


if __name__ == "__main__":
    main()

File: articles/decision-quality-and-strategic-alignment/r/run_all_decision_quality_alignment_workflows.R
# run_all_decision_quality_alignment_workflows.R
# Run all R workflows for Decision Quality and Strategic Alignment.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "decision_quality_strategic_alignment_workflow.R",
  "decision_quality_tables.R",
  "strategic_alignment_tables.R",
  "alignment_drift_tables.R",
  "performance_summary_tables.R",
  "decision_review_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R decision quality and alignment workflows completed.")
